source/grid/tria.cc

```diff
@@ -10228,15 +10228,21 @@ Triangulation<dim, spacedim>::reset_manifold(const types::manifold_id m_number)
   AssertIndexRange(m_number, numbers::flat_manifold_id);
 
   // delete the entry located at number.
-  manifold.erase(m_number);
+  manifold[m_number] =
+    internal::TriangulationImplementation::get_default_flat_manifold<dim,
```

Comment:
Can you document this change of behavior, and same for the `reset_all_manifolds()` function below? I think it would be useful to state that the function doesn't actually remove the entry from the array, but just overwrite them with a flat manifold.

Of course, an even better solution would be to only allow this function to be called if no entity in the triangulation still uses this manifold id. I think that was the intent of the function: To un-register the `manifold_id`.